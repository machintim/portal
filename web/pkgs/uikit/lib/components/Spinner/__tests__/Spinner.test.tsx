import { describe, expect, test } from 'vitest';
import { render } from '@testing-library/react';
import { css } from '@emotion/css';
import SpinnerSvg from '../Spinner';

describe('SpinnerSvg', () => {
  // Snapshot tests

  test('snapshot: default props', () => {
    const renderResult = render(<SpinnerSvg />);
    expect(renderResult.container.firstChild).toMatchSnapshot();
  });

  test('snapshot: custom size', () => {
    const renderResult = render(<SpinnerSvg size={32} />);
    expect(renderResult.container.firstChild).toMatchSnapshot();
  });

  test('Snapshot: custom className', () => {
    const style = css`
      color: #0d68cf;
    `;
    const renderResult = render(<SpinnerSvg className={style} />);
    expect(renderResult.container.firstChild).toMatchSnapshot();
  });

  // Interaction tests

  test('Can be found with default testId', () => {
    const defaultTestId = 'ucl-uikit-spinner';
    const { getByTestId } = render(<SpinnerSvg testId={defaultTestId} />);
    const svg = getByTestId(defaultTestId);
    expect(svg).toBeDefined();
  });

  test('test ID: custom', () => {
    const { getByTestId } = render(<SpinnerSvg testId='custom-test-id' />);
    const svg = getByTestId('custom-test-id');
    expect(svg).toBeDefined();
  });
});
