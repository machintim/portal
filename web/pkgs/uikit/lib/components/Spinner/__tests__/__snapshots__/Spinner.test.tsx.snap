// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SpinnerSvg > Snapshot: custom className 1`] = `
<svg
  class="ucl-uikit-spinner css-1dy6s6p"
  data-testid="ucl-uikit-spinner"
  height="24"
  viewBox="0 0 50 50"
  width="24"
>
  <g
    class="css-1tl81jg"
  >
    <circle
      class="css-lyzf49"
      cx="25"
      cy="25"
      fill="none"
      r="20"
      stroke="currentColor"
      stroke-width="8"
    />
  </g>
</svg>
`;

exports[`SpinnerSvg > snapshot: custom size 1`] = `
<svg
  class="ucl-uikit-spinner"
  data-testid="ucl-uikit-spinner"
  height="32"
  viewBox="0 0 50 50"
  width="32"
>
  <g
    class="css-1tl81jg"
  >
    <circle
      class="css-lyzf49"
      cx="25"
      cy="25"
      fill="none"
      r="20"
      stroke="currentColor"
      stroke-width="8"
    />
  </g>
</svg>
`;

exports[`SpinnerSvg > snapshot: default props 1`] = `
<svg
  class="ucl-uikit-spinner"
  data-testid="ucl-uikit-spinner"
  height="24"
  viewBox="0 0 50 50"
  width="24"
>
  <g
    class="css-1tl81jg"
  >
    <circle
      class="css-lyzf49"
      cx="25"
      cy="25"
      fill="none"
      r="20"
      stroke="currentColor"
      stroke-width="8"
    />
  </g>
</svg>
`;
